src/state_space/mod.rs

```diff
@@ -141,12 +142,17 @@ where
                                 last_synced_block,
                                 "reorg detected, unwinding state changes"
                             );
-                            unwind_state_changes(
-                                state.clone(),
-                                state_change_cache.clone(),
-                                chain_head_block_number,
-                            )
-                            .await?;
+
```

Comment:
Since this logic is duplicated below, can we isolate it into an impl or helper function?

Reply:
Yes there is quite a bit that can be abstracted into functions. I have a few // TODO: refactor throughout the `StateSpaceManager`. If you want to tackle this feel free! Otherwise Ill jump on it shortly.